Yona/Yona/Settings/SettingsViewController.swift

```diff
@@ -108,6 +96,12 @@ class SettingsViewController: UIViewController {
             UIApplication.sharedApplication().keyWindow?.rootViewController =  UINavigationController(rootViewController: welcome)
         }
     }
+    
+    //MARK: Navigation, Segue
+    override func shouldPerformSegueWithIdentifier(identifier: String, sender: AnyObject?) -> Bool {
+        //return false so we can load our detail data before pushing segue
+        return false
+    }
 }
```

Comment:
Will remove all the unwanted code. As I mentioned in the comment, I will do a code refactor on monday.